zerver/management/commands/makemessages.py

```diff
@@ -73,3 +115,84 @@ def my_templatize(src, origin=None):
             trans_real.block_re = old_block_re
             trans_real.templatize = old_templatize
             trans_real.constant_re = old_constant_re
+
+    def extract_strings(self, data):
+        translation_strings = {}
+        for regex in frontend_compiled_regexes:
+            for match in regex.findall(data):
+                translation_strings[match] = ""
+
+        return translation_strings
+
+    def get_translation_strings(self):
+        translation_strings = {}
+        dirname = self.get_template_dir()
+
+        for filename in os.listdir(dirname):
+            if filename.endswith('handlebars'):
+                with open(os.path.join(dirname, filename)) as reader:
+                    data = reader.read()
+                    translation_strings.update(self.extract_strings(data))
+
+        return translation_strings
+
+    def get_template_dir(self):
+        return self.frontend_source
+
+
+    def get_namespace(self):
+        return self.frontend_namespace
+
+    def get_locales(self):
+        locale = self.frontend_locale
+        exclude = self.frontend_exclude
+        process_all = self.frontend_all
+
+        paths = glob.glob('%s/*' % self.default_locale_path,)
+        locale_dirs = list(filter(os.path.isdir, paths))
+        all_locales = list(map(os.path.basename, locale_dirs))
+
+        # Account for excluded locales
+        if process_all:
+            locales = all_locales
+        else:
+            locales = locale or all_locales
+            locales = set(locales) - set(exclude)
+
+        return locales
+
+    def get_base_path(self):
+        return self.frontend_output
+
+    def get_output_paths(self):
+        base_path = self.get_base_path()
+        locales = self.get_locales()
+        for path in [os.path.join(base_path, locale) for locale in locales]:
+            if not os.path.exists(path):
+                os.makedirs(path)
+
+            yield os.path.join(path, self.get_namespace())
+
+    def get_new_strings(self, old_strings, translation_strings):
+        """
+        Missing strings are removed, new strings are added and already
+        translated strings are not touched.
+        """
+        new_strings = {}
+        for k in translation_strings:
+            new_strings[k] = old_strings.get(k, k)
+
+        return new_strings
+
+    def write_translation_strings(self, translation_strings):
+        for locale, output_path in zip(self.get_locales(), self.get_output_paths()):
+            self.stdout.write("[frontend] processing locale {}".format(locale))
+            try:
+                with open(output_path, 'r') as reader:
+                    old_strings = ujson.load(reader)
+            except (IOError, ValueError):
+                old_strings = {}
+
+            new_strings = self.get_new_strings(old_strings, translation_strings)
+            with open(output_path, 'w') as writer:
+                ujson.dump(new_strings, writer)
```

Comment:
I think `ujson` takes the indent argument you used to have here -- probably worth restoring that so we don't have all the translations.json content on one line..